oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyState.java

```diff
@@ -127,8 +188,25 @@ public boolean equals(Object object) {
             return true;
         } else if (object instanceof DocumentPropertyState) {
             DocumentPropertyState other = (DocumentPropertyState) object;
-            return this.name.equals(other.name)
-                    && this.value.equals(other.value);
+            // Compare names and raw un-parsed values
+            if (!this.name.equals(other.name) || !this.getValue().equals(other.getValue())) {
```

Comment:
This will always decompress, no?

Reply:
Not necessary.

Reply:
Yes, but only then, not always.

Reply:
Right, but IIUC the idea was to avoid uncompressing - and that's actually what the code does after this check. So either we don't aim to avoid uncompressing (but then the lines after this aren't necessary), or we want to avoid it (then this condition here needs to be adjusted). Currently we do a bit of both.

Reply:
Done.